doc/manual/src/design/store/entries.md

```diff
@@ -0,0 +1,46 @@
+# Store Entries
+
+File system data in Nix is organized into *store entries*.
+A store entry is the combination of
+
+  - some file system data
+  - references to store entries
+
+## File system data
+
+Nix supports the a similar model of the file system as Git.
+Namely, every file system object falls into these three cases:
+
+ - File: arbitrary data
+
+ - Directory: mapping of names to child file system objects.
+   File children additionally have an executable flag.
+
+ - Symlink: may point anywhere.
+   In particular, Symlinks that do not point within the containing file system data or that of another store entry referenced by the containing store entry are allowed, but might not function as intended.
+
+A bare file as the "root" file system object is allowed.
+Note that there is no containing directory object to store its executable bit; it's deemed non-executable by default.
```

Comment:
I had some trouble understanding this sentence on my first few passes. How about:

"Note that directories describe if a file is executable or not, therefore when a file has no containing directory, it cannot be executable"

It's also worth mentioning that this is not a default, or non-default behavior. The way it's described in the bullets  above, it is impossible for a bare file as a store path to be executable.

Reply:
```suggestion
Note that it cannot be executable, though.
This is a consequence of the executable flags being part of the child entries of the directory, rather than the child files themselves.
A root file has no parent directory; so there is no child entry about the root file, and therefore no executable flag for it.
Without a flag saying which, whether root files are executable or non-executable by default must be decided by convention, and the choice of Nix (and git) is to make them non-executable.
```